### PR TITLE
feat(i18n): I18n used only if that module is installed and initialized

### PR DESCRIPTION
**Pull request by doktordirk**
not 100% sure if new Optional(I18N) is the best way, but i didn't spot an example which used that feature.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -1,19 +1,27 @@
...
-@inject(humane, I18N)
+@inject(humane, new Optional(I18N))
 export class Notification {
   constructor (humane, i18n) {
     this.humane = humane;
     this.i18n   = i18n;
   }
 
+  _log(message, options) {
+    if (this.i18n && this.i18n.i18next.isInitialized())
```
Comment (RWOverdijk) [inline src/index.js]: Curly braces

File: src/index.js [hunk 2]
```diff
@@ -1,19 +1,27 @@
...
 
-@inject(humane, I18N)
+@inject(humane, new Optional(I18N))
 export class Notification {
   constructor (humane, i18n) {
     this.humane = humane;
     this.i18n   = i18n;
   }
 
+  _log(message, options) {
```
Comment (RWOverdijk) [inline src/index.js]: Remove the underscore

**Comment by RWOverdijk** [comment]
If we want to do this, we have to also [remove i18n as a dependency](https://github.com/SpoonX/aurelia-notification/blob/master/package.json#L54).

**Comment by doktordirk** [comment]
right.

**Comment by RWOverdijk** [comment]
@doktordirk Also, thank you. :D

**Comment by doktordirk** [comment]
re. i18n dependency. as it is it has to stay. my code still has/needs `import {I18N} from 'aurelia-i18n';`. Wouldn't work without some aurelia-i18n somewhere.

File: doc/CHANGELOG.md [hunk 3]
```diff
@@ -0,0 +1,14 @@
+<a name"0.0.2"></a>
```
Comment (RWOverdijk) [inline doc/CHANGELOG.md:1]: ? Why is this here?

File: src/index.js [hunk 4]
```diff
@@ -1,19 +1,28 @@
...
 import {inject} from 'aurelia-framework';
 import {I18N} from 'aurelia-i18n';
+import {Optional} from 'aurelia-dependency-injection';
 
-@inject(humane, I18N)
+@inject(humane, new Optional(I18N))
 export class Notification {
-  constructor (humane, i18n) {
-    this.humane = humane;
+  constructor(_humane, i18n) {
```
Comment (RWOverdijk) [inline src/index.js]: Why do you like underscores so much? :p Please take out the underscore :)

File: src/index.js [hunk 5]
```diff
@@ -1,19 +1,28 @@
 import humane from 'wavded/humane-js';
 import {inject} from 'aurelia-framework';
 import {I18N} from 'aurelia-i18n';
+import {Optional} from 'aurelia-dependency-injection';
```
Comment (RWOverdijk) [inline src/index.js]: Remove this

File: src/index.js [hunk 6]
```diff
@@ -1,19 +1,28 @@
...
 
-@inject(humane, I18N)
+@inject(humane, new Optional(I18N))
```
Comment (RWOverdijk) [inline src/index.js]: Remove new Optional(

File: src/index.js [hunk 7]
```diff
@@ -1,19 +1,28 @@
...
-    this.humane = humane;
+  constructor(_humane, i18n) {
+    this.humane = _humane;
     this.i18n   = i18n;
   }
 
-  success (message) {
-    this.humane.log(this.i18n.tr(message), {addnCls: 'success'});
+  log(message, options) {
+    if (this.i18n && this.i18n.i18next.isInitialized()) {
```
Comment (RWOverdijk) [inline src/index.js]: Remove `this.i18n &&`; but keep `this.i18n.i18next.isInitialized()`. That way, we only use it if it's configured and it automatically works.

**Comment by doktordirk** [comment]
besides the old gulp lint complaining about humane being previously defined, i also dont like to use humane for the class and the instance. hmm, but i could try import humane as Humane or so. but then, got a new version already which might need underscores. 
as for <a name"0.0.2"></a>. those are added by conventional-changelog >= 0.0.13. , doesn't seem to do much harm
re gitter: i'll quickly add my (not too much changed last version in case it'll help)

File: src/index.js (hunk 4 above)
Comment (RWOverdijk) [inline src/index.js]: That's fair; import as Humane.

File: src/index.js (hunk 7 above)
Comment (doktordirk) [inline src/index.js]: doesn't work. Maybe it's just a bug in aurelia-i18n but as of yesterday it crashes on the new i18n() in inject(I18n) if i18n wasn't setup in main.js  with at least .plugin('aurelia-i18n', () => {})

**Comment by doktordirk** [comment]
about injecting i18n maybe see https://github.com/aurelia/i18n/issues/48#issuecomment-169722881

**Comment by doktordirk** [comment]
other than the unresolved i18n bug causing the Optional(I18n) i better be finished for now
